spec/cell_spec.rb

```diff
@@ -8,69 +8,107 @@
       cell = Cell.new("B4")
       expect(cell).to be_a Cell
     end
+
     it 'takes a coordinate' do
       cell = Cell.new("B4")
       expect(cell.coordinate).to eq("B4")
     end
   end
 
-  describe 'test methods' do
-    it 'empty? returns boolean' do
+  describe '#empty?' do
+    it 'returns true when empty' do
       cell = Cell.new("B4")
       expect(cell.empty?).to be true
+    end
 
+    it 'returns false when cell is occupied' do
+      cell = Cell.new("B4")
       cruiser = Ship.new("Cruiser", 3)
       cell.place_ship(cruiser)
       expect(cell.empty?).to be false
     end
+  end
 
-    it 'ship returns the ship object' do
+  describe '#ship' do
+    it 'returns nil when the ship is not in the cell' do
       cell = Cell.new("B4")
       expect(cell.ship).to eq nil
+    end
 
+    it 'returns the ship object when it is in the cell' do
+      cell = Cell.new("B4")
       cruiser = Ship.new("Cruiser", 3)
       cell.place_ship(cruiser)
       expect(cell.ship).to eq(cruiser)
     end
+  end
 
-    it 'fired_upon? returns boolean' do
+  describe '#fired_upon?' do
+    it 'returns false when the cell has been fired upon' do
       cell = Cell.new("B4")
-
       cruiser = Ship.new("Cruiser", 3)
       cell.place_ship(cruiser)
       expect(cell.fire_upon?).to be false
+    end
 
+    it 'returns true when the cell has been fired upon' do
+      cell = Cell.new("B4")
+      cruiser = Ship.new("Cruiser", 3)
       cell.fire_upon
       expect(cell.fired_upon?).to be true
     end
+  end
 
-    it 'render gives a visual status of cells' do
+  describe '#render' do
+    it 'displays . when the cell has not been fired upon' do
       cell_1 = Cell.new("B4")
       cell_2 = Cell.new("B2")
       cell_3 = Cell.new("B3")
-
       cruiser = Ship.new("Cruiser", 3)
       cell_1.place_ship(cruiser)
-
       expect(cell_1.render).to eq(".")
       expect(cell_2.render).to eq(".")
       expect(cell_3.render).to eq(".")
+    end
 
+    it 'displays S in the cell when the optional argument is used' do
+      cell_1 = Cell.new("B4")
+      cell_2 = Cell.new("B2")
+      cell_3 = Cell.new("B3")
+      cruiser = Ship.new("Cruiser", 3)
+      cell_1.place_ship(cruiser)
       expect(cell_1.render(true)).to eq("S")
+    end
 
+    it 'displays H when a ship has been hit in the cell' do
+      cell_1 = Cell.new("B4")
+      cell_2 = Cell.new("B2")
+      cell_3 = Cell.new("B3")
+      cruiser = Ship.new("Cruiser", 3)
       cell_1.fire_upon
       expect(cell_1.fired_upon?).to be true
       expect(cell_1.render).to eq("H")
+    end
 
+    it 'displays M when a cell is fired upon but is a miss' do
+      cell_1 = Cell.new("B4")
+      cell_2 = Cell.new("B2")
+      cell_3 = Cell.new("B3")
```

Comment:
in this block we don't use cell_1 and cell_3, so we should delete those

Reply:
A good catch and we can update the naming of the variable so we don't start at cell_2.